tests/python_package_test/test_sklearn.py

```diff
@@ -1301,3 +1302,26 @@ def test_multiclass_custom_objective():
     np.testing.assert_allclose(builtin_obj_preds, custom_obj_preds, rtol=0.01)
     assert not callable(builtin_obj_model.objective_)
     assert callable(custom_obj_model.objective_)
+
+
+def test_negative_n_jobs():
+    n_threads = joblib.cpu_count()
+    if n_threads <= 1:
+        return None
+    val_minus_two = n_threads - 1
```

Comment:
```suggestion
    val_minus_two = n_threads - 2
```

Reply:
Joblib's formula is `n_cpus + 1 + n_jobs`. Thus, passing `-2` means "use all but one thread".

Reply:
Please add an inline comment explaining this. Right this line is extremely confusing and can be read as "val_minus_two = val - 1" which looks like a bug.

Reply:
I mean to add a comment for very confusing `val_minus_two = n_threads - 1` line exactly, not for `n_jobs=-2`.